system-contracts/contracts/libraries/SystemContractsCaller.sol

```diff
@@ -114,6 +112,28 @@ library SystemContractsCaller {
                 success := call(msgValueSimulator, callAddr, value, to, farCallAbi, forwardMask, 0)
             }
         }
+        if (success) {
```

Comment:
not everyone needs to copy the entire returndata rightaway Please move this logic for copying the returndata into the simulator since it is the only one that needs it

Reply:
added flag to specify returndata mode